lib/src/helpers/path.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+// For the most part, Sass treats URIs and paths interchangeably, but Node does
+// not. These helper methods smooth over Node's URI/path dichotomy.
+
+import * as p from 'path';
+import {fileURLToPath, pathToFileURL, URL} from 'url';
+
+/**
+ * Returns the URL that represents `pathOrUri`.
+ *
+ * If `pathOrUri` is an absolute file path, returns a `file:` UTL.
+ * If `pathOrUri` is a URL, returns the URL.
+ * If `pathOrUri` is relative, returns a relative URI.
+ */
+export function toUrl(pathOrUri: string): string {
+  try {
+    return new URL(pathOrUri).toString();
+  } catch (error) {
+    if (p.isAbsolute(pathOrUri)) {
+      return pathToFileURL(pathOrUri).toString();
+    } else {
+      return pathOrUri;
```

Comment:
Even relative paths aren't interchangeable with URLs, especially on Windows (but in edge cases also on POSIX systems). Unfortunately it looks like there's not an idiomatic way to do this transformation in Node, but you get 80% of the way there by converting backslashes to forward slashes on Windows.

Reply:
Ah, got it. I understand the Windows case, but shouldn't a relative POSIX path be interchangeable with a relative URL?

Reply:
No, for two reasons:

1. POSIX paths allow effectively any character (technically, byte sequence) to exist unescaped, while URLs only allow a subset of ASCII.
2. The `%` character in particular is a literal character in POSIX paths but an escape character in URLs, so for URLs `f%6Fo` is equivalent to `foo` but for POSIX paths they're totally different.